File: collect_app/src/main/java/org/odk/collect/android/widgets/utilities/AudioPlayer.java
package org.odk.collect.android.widgets.utilities;

import org.odk.collect.audioclips.Clip;

import java.util.List;
import java.util.function.Consumer;

public interface AudioPlayer {

    void play(Clip clip);

    void pause();

    void setPosition(String clipId, Integer position);

    void onPlayingChanged(String clipID, Consumer<Boolean> playingConsumer);

    void onPositionChanged(String clipID, Consumer<Integer> positionConsumer);

    void onPlaybackError(Consumer<Exception> error);

    void stop();

    void playInOrder(List<Clip> clips);
}

File: audio-clips/src/main/java/org/odk/collect/audioclips/AudioPlayer.kt
package org.odk.collect.audioclips

import androidx.lifecycle.LiveData
import java.util.function.Consumer

interface AudioPlayer {
    fun isLoading(): LiveData<Boolean>

    fun play(clip: Clip)

    fun pause()

    fun setPosition(clipId: String, position: Int)

    fun onPlayingChanged(clipID: String, playingConsumer: Consumer<Boolean>)

    fun onPositionChanged(clipID: String, positionConsumer: Consumer<Int>)

    fun onPlaybackError(error: Consumer<Exception>)

    fun stop()

    fun playInOrder(clips: List<Clip>)
}
